src/reporting-config/reporting-config-parsers.ts

```diff
@@ -1,86 +1,249 @@
-import { ReportingConfigApiResponse } from '../api';
-import { IndexedReportingConfig, NormalizedReportingConfig } from './types';
+import { ApiFeatureGroups, ApiFeatures, ApiTasks, ReportingConfigApiResponse } from '../api';
+import {
+	FeatureGroups,
+	FeatureParentEntityType,
+	Features,
+	IndexedReportingConfig,
+	NormalizedReportingConfig,
+	Products,
+	TaskMapping,
+	TaskParentEntityType,
+	Tasks,
+} from './types';
 
 // Defining all the core logic for how we normalize and index reporting configs here.
 // We are keeping this separate from how this logic is used in reducers and thunks.
 // This allows easy unit testing of this critical logic apart from how its consumed.
 
 export function normalizeReportingConfig(
```

Comment:
The initial change here was the need to also normalize tasks as their own entities. After doing that, this function was a bit of a big logical mess and needed some refactoring. So I refactored it to make it a bit more modularized.

It's probably easiest to just look at the final version of the function in the branch and see if it makes sense.